ocaml/database/block_device_io.ml

```diff
@@ -515,11 +516,11 @@ let action_writedelta block_dev_fd client datasock target_response_time =
   let generation_count = read_generation_count client in
   read_separator client;
   let length_str = read_length client in
-  let length = int_of_string length_str in
+  let length = int_of_string (Bytes.unsafe_to_string length_str) in
```

Comment:
This is just an idea: We could adopt a programming style where we override the previous let binding containing the `bytes` with a new one in these cases, so that we know for sure that the `bytes` is not accessible any more and therefore cannot be mutated:
```
   let length = read_length client in
   let length = int_of_string (Bytes.unsafe_to_string length) in
```